ocaml/xapi/xapi_vdi.ml

```diff
@@ -861,6 +861,35 @@ let export_changed_blocks ~__context ~vdi_from ~vdi_to =
   Storage_access.transform_storage_exn
     (fun () ->
        C.VDI.export_changed_blocks ~dbg:(Ref.string_of task) ~sr ~vdi_from ~vdi_to
-    );
+    )
+
+let get_nbd_info ~__context ~self =
+  if (Db.VDI.get_type ~__context ~self) = `cbt_metadata then begin
+    error "VDI.get_nbd_info: called with a VDI of type cbt_metadata (at %s)" __LOC__;
+    raise (Api_errors.Server_error (Api_errors.vdi_incompatible_type, [ Ref.string_of self; Record_util.vdi_type_to_string `cbt_metadata ]))
+  end;
+
+  let sr = Db.VDI.get_SR ~__context ~self in
+  let hosts_with_attached_pbds =
```

Comment:
The code to populate `hosts_with_attached_pbds` looks correct.
Using one of the `get_..._where` functions might be faster but this is not going to be called often so it doesn't matter.